### DISCUSSION TITLE
Animating GeolocateControl circle with trackUserLocation true

### DISCUSSION THREAD

**Discussion by ugnius-s** [maplibre/maplibre-gl-js#5257] [Ideas]
Hello, thank you for supporting this great library.

I'm working on a project and one of the users asked a feature to improve the user experience.

As of now when we show the user location with tracking enabled the position changes are not being animated and are choppy. Which for some drivers it causes minor visual annoyance. Let me give you and example:

https://github.com/user-attachments/assets/0fd92de3-d42d-486f-8806-0bb3e091cda4

Is there maybe something in plans when this can be done? Or if someone would guide me where to start to look I can implement this myself. And open a pull request to contribute.

Thank you

**Comment by HarelM** [comment]
All the code related to this control can be found here: 
https://github.com/maplibre/maplibre-gl-js/blob/main/src/ui/control/geolocate_control.ts

It's not too complicated, feel free to take a look. 
I generally think there's a need to separate the UI from the state management and from the "Hardware" code to allow more flexibility.

**Reply by ugnius-s** [reply]
Thanks. Will check how it is right now and what can be done.
